crates/red_knot_python_semantic/src/types.rs

```diff
@@ -588,6 +625,16 @@ pub struct ClassType<'db> {
 }
 
 impl<'db> ClassType<'db> {
+    /// Return true if this class is the builtin type with given name.
+    #[allow(unused)]
+    pub(crate) fn is_builtin_named(self, db: &'db dyn Db, name: &str) -> bool {
+        name == self.name(db).as_str()
+            && file_to_module(db, self.body_scope(db).file(db))
+                // Builtin module names are special-cased in the resolver, so there can't be a
+                // module named builtins other than the actual builtins.
+                .is_some_and(|module| module.name().as_str() == "builtins")
```

Comment:
Doesn't `name` have a `PartialEq<str>` implementation

```suggestion
                .is_some_and(|module| module.name() == "builtins")
```